progenitor-impl/src/cli.rs

```diff
@@ -0,0 +1,482 @@
+                        .required(#required)
+                        .value_parser(clap::value_parser!(#arg_type_name))
+                        #help
+                }
+            });
+
+        let maybe_body_param = method.params.iter().find(|param| {
+            matches!(&param.kind, OperationParameterKind::Body(_))
+                // TODO not sure how to deal with raw bodies right now
+                    && matches!(&param.typ, OperationParameterType::Type(_))
+        });
+
+        let body_arg = maybe_body_param.map(|param| {
+            let OperationParameterType::Type(type_id) = &param.typ else {
+                    unreachable!();
+                };
+
+            let body_args = self.type_space.get_type(type_id).unwrap();
+
+            let body_arg = match body_args.details() {
+                typify::TypeDetails::Struct(s) => {
+                    s.properties_info()
+                        .filter_map(|prop_info| {
+                            let TypeStructPropInfo {
+                                name: prop_name,
+                                description,
+                                required,
+                                type_id: prop_type_id,
+                            } = prop_info;
+                            let prop_type = self
+                                .type_space
+                                .get_type(&prop_type_id)
+                                .unwrap();
+                            let prop_type_ident = prop_type.ident();
+                            let good =
+                                prop_type.has_impl(TypeSpaceImpl::FromStr);
+                            // assert!(good || !required);
+
+                            good.then(|| {
+                                let help =
+                                    description.as_ref().map(|description| {
+                                        quote! {
+                                            .help(#description)
+                                        }
+                                    });
+                                quote! {
+                                    clap::Arg::new(#prop_name)
+                                        .long(#prop_name)
+                                        .required(#required)
+                                        .value_parser(clap::value_parser!(
+                                            #prop_type_ident
+                                        ))
+                                        #help
+                                }
+                            })
+                        })
+                        .collect::<Vec<_>>()
+                }
+                _ => Vec::new(),
+            };
+
+            quote! {
+                #(
+                    .arg(#body_arg)
+                )*
+            }
+        });
+
+        // TODO parameter for body as input json (--body-input?)
+        // TODO parameter to output a body template (--body-template?)
```

Comment:
By body template do you mean a table?

Reply:
I'd like to have a parameter that outputs "blank" JSON output so that the user can output that to a file, edit it to fill in appropriate values, and then input it to another option that accepts a JSON-formatted file as input.